src/meta/MetaServiceUtils.cpp

```diff
@@ -32,6 +32,47 @@ nebula::cpp2::ErrorCode backupTable(kvstore::KVStore* kvstore,
                std::make_move_iterator(backupTableFiles.end()));
   return nebula::cpp2::ErrorCode::SUCCEEDED;
 }
+
+bool isLegalTypeConversion(cpp2::ColumnTypeDef from, cpp2::ColumnTypeDef to) {
+  // If not type change, return true. Fixed string will be handled separately.
+  if (from.get_type() == to.get_type() &&
+      from.get_type() != nebula::cpp2::PropertyType::FIXED_STRING) {
+    return true;
+  }
+  // For unset type, always return true
+  if (from.get_type() == nebula::cpp2::PropertyType::UNKNOWN) {
+    return true;
+  }
+  // fixed string can be converted to string or wider fixed string
+  if (from.get_type() == nebula::cpp2::PropertyType::FIXED_STRING) {
+    if (to.get_type() == nebula::cpp2::PropertyType::STRING) {
+      return true;
+    } else if (to.get_type() == nebula::cpp2::PropertyType::FIXED_STRING) {
+      return from.get_type_length() <= to.get_type_length();
+    } else {
+      return false;
+    }
+  }
+  // int is only allowed to convert to wider int
+  if (from.get_type() == nebula::cpp2::PropertyType::INT32) {
+    return to.get_type() == nebula::cpp2::PropertyType::INT64;
+  }
+  if (from.get_type() == nebula::cpp2::PropertyType::INT16) {
+    return to.get_type() == nebula::cpp2::PropertyType::INT64 ||
+           to.get_type() == nebula::cpp2::PropertyType::INT32;
+  }
+  if (from.get_type() == nebula::cpp2::PropertyType::INT8) {
+    return to.get_type() == nebula::cpp2::PropertyType::INT64 ||
+           to.get_type() == nebula::cpp2::PropertyType::INT32 ||
+           to.get_type() == nebula::cpp2::PropertyType::INT16;
+  }
+  // Float is only allowed to convert to double
+  if (from.get_type() == nebula::cpp2::PropertyType::FLOAT) {
+    return to.get_type() == nebula::cpp2::PropertyType::DOUBLE;
+  }
+  // Forbid all the other conversion, as the old data are too different from the new data.
```

Comment:
shall we forbid the type conversion from boolean to int?

Reply:
Data are still correct if we allow for bool <-> int conversion. But the data before conversion and after conversion will look different. In a strict manner, we can disallow any conversion that will make data look different. Or if we are not that strict, we can allow bool <-> int conversion. Please clarify. cc. @MuYiYong @critical27

Reply:
prefer strict

Reply:
I think we could forbid it this version I think, it will introduce extra work in storage.